gradio/components/chatbot.py

```diff
@@ -139,24 +151,37 @@ def __init__(
         self.placeholder = placeholder
 
     def _preprocess_chat_messages(
-        self, chat_message: str | FileMessage | None
-    ) -> str | tuple[str | None] | tuple[str | None, str] | None:
+        self,
+        chat_message: str
+        | FileMessage
+        | List[FileMessage]
+        | GalleryData
+        | PlotData
+        | None,
+    ) -> str | GradioComponent | tuple[str | None] | tuple[str | None, str] | None:
         if chat_message is None:
             return None
         elif isinstance(chat_message, FileMessage):
             if chat_message.alt_text is not None:
                 return (chat_message.file.path, chat_message.alt_text)
             else:
                 return (chat_message.file.path,)
-        elif isinstance(chat_message, str):
+        elif isinstance(chat_message, (str)):
             return chat_message
+        elif isinstance(chat_message, GalleryData):
+            value = Gallery().preprocess(chat_message)
```

Comment:
I would set `value='filepath`` explicitly here. Cause if we ever change the default it would increase the latency a lot by converting to PIL or whatever.